python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -133,50 +144,73 @@ def run_iteration(
         self,
         workspace_process_context: IWorkspaceProcessContext,
     ) -> TDaemonGenerator:
+        run_coordinator = workspace_process_context.instance.run_coordinator
+        if not isinstance(run_coordinator, QueuedRunCoordinator):
+            check.failed(f"Expected QueuedRunCoordinator, got {run_coordinator}")
+
+        run_queue_config = run_coordinator.get_run_queue_config()
+
         instance = workspace_process_context.instance
-        runs_to_dequeue = self._get_runs_to_dequeue(instance)
-        yield from self._dequeue_runs_iter(workspace_process_context, runs_to_dequeue)
+        runs_to_dequeue = self._get_runs_to_dequeue(instance, run_queue_config)
+        yield from self._dequeue_runs_iter(
+            workspace_process_context, run_coordinator, runs_to_dequeue, run_queue_config
+        )
 
     def _dequeue_runs_iter(
         self,
         workspace_process_context: IWorkspaceProcessContext,
+        run_coordinator: QueuedRunCoordinator,
         runs_to_dequeue: List[DagsterRun],
+        run_queue_config: RunQueueConfig,
     ):
-        run_coordinator = workspace_process_context.instance.run_coordinator
-        if not isinstance(run_coordinator, QueuedRunCoordinator):
-            check.failed(f"Expected QueuedRunCoordinator, got {run_coordinator}")
-
         if run_coordinator.dequeue_use_threads:
             yield from self._dequeue_runs_iter_threaded(
                 workspace_process_context,
                 runs_to_dequeue,
                 run_coordinator.dequeue_num_workers,
+                run_queue_config,
             )
         else:
             yield from self._dequeue_runs_iter_loop(
                 workspace_process_context,
                 runs_to_dequeue,
+                run_queue_config,
             )
 
+    def _dequeue_run_thread(
+        self,
+        workspace_process_context: IWorkspaceProcessContext,
+        run: DagsterRun,
+        run_queue_config: RunQueueConfig,
+    ) -> bool:
+        return self._dequeue_run(
+            workspace_process_context.instance,
+            workspace_process_context.create_request_context(),
```

Comment:
curious how intentional it is that we make a new request context on each thread

Reply:
Im not sure how intentional it was. 

When the thread pool executor work queue is deeper than the number of workers, you can end up accumulating request contexts in memory which is a problem we have had before.

I don't believe creating request contexts has shown up as costly in profiling.